src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -821,4 +846,81 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         }
     }
 
+
+    public static class AnonymizationSettings {
+        // TODO:  Provide a way to modify these
+        private Set<String> excludedWords = new HashSet<>();
+
+        private boolean anonymizeLabels;
+        private boolean anonymizeItems;
+        private boolean anonymizeViews;
+        private boolean anonymizeNodes;
+        private boolean anonymizeComputers;
+        private boolean anonymizeUsers;
+
+        public AnonymizationSettings() {
+            // TODO:  Where should this actually go?  Here?  Elsewhere?  I don't think that these particular strings
+            // should be modifiable by users
+            excludedWords.add("a");
+            excludedWords.add("the");
+            excludedWords.add("all");
+            excludedWords.add("master");
+            excludedWords.add("jenkins");
+            excludedWords.add("node");
+            excludedWords.add("computer");
+            excludedWords.add("item");
+            excludedWords.add("label");
+            excludedWords.add("view");
+            excludedWords.add("user");
+            excludedWords.add("");
```

Comment:
Some of these seem weird to me (e.g., "a", "the", "all") and there are some words that seem to be related to what's here but are not included (e.g. "agent", "slave", "job", "build"). Are these defaults a just-in-case defense, or are they necessary for the feature to work correctly?

Reply:
I guess the issue is that if someone created a job called "computer", then all uses of "computer" would get anonymized, even where it isn't being used as the name of a job?

Reply:
Yeah, exactly.  I added some words that I know are going to be used a lot and that are extremely unlikely to pose any sort of risk to not be anonymized.  

Unfortunately, I couldn't figure out a better way to do this.  I don't think this is good, I just don't know what the other options are short of rewriting how the logs are written in the first place (which is rather unfeasible).

Reply:
The common terminology for these are _stop words_. [Example list](https://kb.yoast.com/kb/list-stop-words/).

Reply:
> A more appropriate place might be a text file resource of some sort. That way end users could potentially add more things if necessary locally.

I think that what we have here should cover most cases (I would probably leave out the normal English words and just focus on the Jenkins model objects), but to be more exhaustive we could also add some dynamic entries based on iterating through the result of `ExtensionList.lookup(AbstractItem.class)` and adding `AbstractItem#getPronoun` and `AbstractItem#getTaskNoun`.

Reply:
Definitely like that idea. I'm currently trying to make the set of strings configurable in the settings as well.